src/server/object_services/map_reader.js

```diff
@@ -44,7 +46,8 @@ async function read_object_mapping(obj, start, end, location_info) {
         end_gt: rng.start,
     });
     // console.log('TODO GGG read_object_mapping', parts);
-    const chunks = await read_parts_mapping(parts);
+    const chunks = await read_parts_mapping(parts, location_info);
+    update_chunks_on_read(chunks, location_info);
```

Comment:
@jackyalbo according to your explanation we should `await` here (although I don't agree with that)